metabox/metabox/core/machine.py

```diff
@@ -286,6 +286,17 @@ def _get_install_dependencies_cmds(self):
             # Use <21 to get the latest 20 as 21+ is not supported here
             return [
                 "bash -c 'sudo python3 -m pip install -U \"pip<21\"'",
+                (
+                    # This is here because this pip version does not support
+                    # installing dependencies from pyproject.toml
+                    "bash -c 'sudo python3 -m pip install "
```

Comment:
1. This cannot be done this way. I know that for you it'll be easy to understand why it's here and how to extend this with time, but for anybody else it will be a hassle.
2. The list is different to the one in pyproject.toml, kind of defeating the whole point of this PR. We want to verify if all the deps are there in "production" so for this test to make sense this should do the same thing as "production" build would do. I think this is not the case here.

The first two ways to solve this that popped up in my brain are:
1. metabox has it's own python environment, that is not ancient, thus it knows how to read the deps from the toml. This can be forwarded by MB itself.
2.
 ```import toml; print('\n'.join(toml.load("pyproject.toml").get('project', {}).get('dependencies', {})))```

Reply:
The list content is (should, I have double checked this but I may still be wrong) the same as the pyproject.toml files (I have merged them into one, checkbox-support + checkbox-ng).

The forced versions are sadly here because there is no good automated (sane) way of getting something like this out of pip. If you tell pip to install a package it will always install the latest version that is compatible with your constraint. For example we say `Jinja2>=2.7` and that will make pip install any version above the one we have asked (including `3.1.2`, which is not compatible with python3.5).

Reply:
Right! So let's use python environment that perfectly replicates this, i.e. python3.5-pip.
By doing any intelligent work expanding this list yourself we're cutting away the value of the whole story.

Reply:
Reporting Meet discussion: As above, we agreed that cleaning this up is possibly very complicated and outside the scope of this pr